plugins/catalog-react/src/components/EntityTagPicker/EntityTagPicker.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useMemo } from 'react';
+import {
+  Checkbox,
+  List,
+  ListItem,
+  ListItemText,
+  makeStyles,
+  Theme,
+  Typography,
+} from '@material-ui/core';
+import { Entity } from '@backstage/catalog-model';
+import { EntityTagFilter } from '../../types';
+import { useEntityListProvider } from '../../hooks/useEntityListProvider';
+
+const useStyles = makeStyles<Theme>(theme => ({
+  title: {
+    margin: theme.spacing(1, 0, 0, 1),
+    textTransform: 'uppercase',
+    fontSize: 12,
+    fontWeight: 'bold',
+  },
+  checkbox: {
+    padding: theme.spacing(0, 1, 0, 1),
+  },
+}));
+
+export const EntityTagPicker = () => {
+  const classes = useStyles();
+  const { updateFilters, backendEntities, filters } = useEntityListProvider();
+  const availableTags = useMemo(
+    () => [
+      ...new Set(
+        backendEntities
+          .flatMap((e: Entity) => e.metadata.tags)
+          .filter(Boolean) as string[],
+      ),
+    ],
+    [backendEntities],
+  );
+
+  if (!availableTags.length) return null;
+
+  const onClick = (tag: string) => {
+    const tags = filters.tags?.values ?? [];
+    const newTags = tags.includes(tag)
+      ? [...tags.filter((t: string) => t !== tag)]
+      : [...tags, tag];
+    updateFilters({
+      tags: newTags.length ? new EntityTagFilter(newTags) : undefined,
+    });
+  };
+
+  return (
+    <>
+      <Typography variant="subtitle2" className={classes.title}>
+        Tags
+      </Typography>
+      <List disablePadding dense>
```

Comment:
Hi, I've noticed through our usage that this list can grow in height infinitely as more tags are defined in the catalog causing the entire page to scroll. Is it possible to set some max height on this list so it has its own dedicated scroll bar on overflow? 

or perhaps a design more like the checkbox select list from the api-docs plugin (which I personally think looks cleaner): 
![image](https://user-images.githubusercontent.com/6998196/119030555-0968a880-b978-11eb-9def-7d118aae35e3.png)

also wanted to note that, the api-docs filter also suffers from the too many tags height issue and work in https://github.com/backstage/backstage/pull/5756 might help with resolving the issue on a single shared base component both plugins can reuse

Reply:
checkbox select plus quick filter input box at the top = chef's kiss

Reply:
Yep, great call. I'll update this.

Reply:
Updated to use multi-select autocomplete:

![multi-select-autocomplete](https://user-images.githubusercontent.com/556258/119211946-8f691a00-ba72-11eb-93e8-2876418eba2a.gif)